### PR TITLE
fix: use current binary directory mariadb shared object and dont override env variable

### PR DESCRIPTION
**Pull request by EmosewaMC**
we were overwriting the users environment variable with the build directory always, resulting in inis always being placed in the incorrect spot if a preset was specified, or if the user had an environment variable where they stored the inis.

Also, assuming it passes ci, made it so we use the shared library in the binary directory as opposed to the one

mac please

### CODE REVIEW

**Review by aronwk-aaron — APPROVED** [review]

**Review by Xiphoseer — APPROVED** [review]
Thank you!

File: dDatabase/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,12 @@ add_subdirectory(CDClientDatabase)
 add_subdirectory(GameDatabase)
 
 add_library(dDatabase STATIC "MigrationRunner.cpp")
+
+add_custom_target(conncpp_dylib
+	${CMAKE_COMMAND} -E copy $<TARGET_FILE:MariaDB::ConnCpp> ${PROJECT_BINARY_DIR})
```
Comment (jadebenn) [inline dDatabase/CMakeLists.txt:7]: This is a bit of kludge. It's being placed in the correct directory for a multi-config generator. The problem is our scripts aren't set up to handle that correctly. We shouldn't need this.

File: tests/dCommonTests/CMakeLists.txt [hunk 2]
```diff
@@ -17,18 +17,13 @@ list(APPEND DCOMMONTEST_SOURCES ${DENUMS_TESTS})
...
-add_custom_command(TARGET dCommonTests POST_BUILD
-	COMMAND otool ARGS -l dCommonTests
-	COMMAND otool ARGS -L dCommonTests
-	COMMAND ls
-	COMMAND otool ARGS -D libmariadbcpp.dylib
-	COMMAND install_name_tool ARGS -change libmariadbcpp.dylib @rpath/libmariadbcpp.dylib dCommonTests
-	COMMAND otool ARGS -L dCommonTests
-	WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY})
+	add_custom_target(dCommonTestsLink
+		${CMAKE_COMMAND} -E copy $<TARGET_FILE:MariaDB::ConnCpp> ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (jadebenn) [inline tests/dCommonTests/CMakeLists.txt:24]: Ditto. This is a kludge.

**Review by jadebenn — COMMENTED** [review]
I can't exactly do a 'changes required' since this has already been merged, but I think we need  to revisit this PR.

File: dDatabase/CMakeLists.txt (hunk 1 above)
Comment (EmosewaMC) [reply dDatabase/CMakeLists.txt:7]: It is not being placed in the correct directory for milti-config generators as for mac CI and development, the file is requires to be in the linking directory and the binary directory at these times, and in neither was this true before.

**Pull request merged**